portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java

```diff
@@ -27,18 +27,22 @@
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet responsible for deleting comments. */
-@WebServlet("/delete-data")
+@WebServlet("/delete-comment")
 public class DeleteCommentServlet extends HttpServlet {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment");
+    long id = Long.parseLong(request.getParameter("id"));
+
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment");
     PreparedQuery results = datastore.prepare(query);
 
     for (Entity entity : results.asIterable()) {
-      Key key = entity.getKey();
-      datastore.delete(key);
+      if(id == entity.getKey().getId()) {
+        Key key = entity.getKey();
+        datastore.delete(key);
```

Comment:
Might want to add a break statement once you've deleted the comment to get out of the loop